src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,70 @@
+package frc.robot.subsystems;
+
+//Imports
+import com.ctre.phoenix6.controls.MotionMagicVelocityVoltage;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+
+import static frc.robot.subsystems.ArmSubsystemConfig.canconderconfigs;
+import static frc.robot.subsystems.ArmSubsystemConfig.talonfxconfigs;
+
+
+public class ArmSubsystem extends SubsystemBase {
+
+    private final TalonFX armKraken;
+    private final CANcoder armEncoder;
+    final MotionMagicVoltage motionMagic;
+
+    //Constructor
+    public ArmSubsystem() {
+        armKraken = new TalonFX(ArmSubsystemConfig.ARM_KRAKEN_ID, Constants.CANIVORE_NAME);
+        armEncoder = new CANcoder(ArmSubsystemConfig.ARM_CANCODER_ID, Constants.CANIVORE_NAME);
+
+        armKraken.getConfigurator().apply(talonfxconfigs);
+        armEncoder.getConfigurator().apply(canconderconfigs);
+
+        motionMagic = new MotionMagicVoltage(0);
+        final MotionMagicVelocityVoltage ArmRequest = new MotionMagicVelocityVoltage(0);
+
+    }
+
+
+    @Override
+    public void periodic(){
+
+    }
+
+    public double getEnc() {
+        return armEncoder.getAbsolutePosition().getValue().magnitude();
+    }
+
+    private void moveUp(double speed) {
+        armKraken.set(Math.abs(speed));
+    }
+
+    private void moveDown(double speed) {
+        armKraken.set(-Math.abs(speed));
+    }
+
+    public Command moveUpAndStop(double speed) {
+        return startEnd(() -> moveUp(speed), this::stop);
+    }
+
+    public Command moveDownAndStop(double speed){
+        return startEnd(() -> moveDown(speed), this::stop);
+    }
+
+    //Deploy Arm
+    public Command deployArm(double newPosition){
+        armKraken.setControl(motionMagic.withPosition(newPosition));
+        return deployArm(newPosition);
```

Comment:
um this is an infinite recursive call because it is returning the method itself. instead this should return a runOnce that runs the setControl
```suggestion
return runOnce(() -> armKraken.setControl(motionMagic.withPosition(newPosition)));
```